### DISCUSSION TITLE
Porepy on Windows

### DISCUSSION THREAD

**Discussion by rafaelmarch3** [pmgbergen/porepy#529] [Ideas]
Hello, 

I decided to try to install porepy in a Conda environment on Windows, and I was positively surprised it worked very well. The steps were:

conda create --name porepy python=3.7 
git clone https://github.com/pmgbergen/porepy.git
cd porepy
pip install .

Everything was properly installed in the Conda environment. When you try to import porepy, though, you get an error. There is a necessary "hack", which is to change the following line in the logging.py file:

`path_length = __file__.split("/").index("porepy")`

to 

`path_length = __file__.split("\\").index("porepy")`

After this change, I'm able to run the example and everything works fine. Is it something you would like to update in the INSTALL documentation? It is said there you need a virtual machine or a docker image to get it to work on Windows, but it is actually way easier than that.

Regards,
Rafael.

**Comment by keileg** [comment]
Hi Rafael,

It is correct that install on Windows (and Mac, I have been told) currently is easier than advertised - actually the fix on the path_length is included in a pull request opened today (#527). The reasons we do not flag Windows compatibility higher are

1) we don't want to commit to not breaking the compatibility at a later point (my understanding is that once we introduce a non-trivial dependency which is not available through `conda`, installation on Windows becomes tricky),
2) We have neither the knowledge nor the capacity to deal with another set of the various OS-related issues that pop up once in a while.

That being said, you have a point that the current install instructions are more intimidating than there is reason for. I will update.

Cheers,
Eirik

**Reply by keileg** [reply]
Implemented in #530. Thanks!

**Reply by IvarStefansson** [reply]
Actuually, this should be remedied by #527.
